ERCS/erc-7579.md

```diff
@@ -0,0 +1,351 @@
+
+To solve these problems, we need to standardize the core interfaces for smart accounts and modules. However, it is highly important that this standardization is done with minimal impact on the implementation logic of the accounts, so that smart account vendors can continue to innovate, while also allowing a flourishing, multi-account-compatible module ecosystem.
+
+The goal of this standard is to define the smart account and module interfaces and behavior that is as minimal as possible while ensuring interoperability between accounts and modules.
+
+## Specification
+
+The key words "MUST", "MUST NOT", "REQUIRED", "SHALL", "SHALL NOT", "SHOULD", "SHOULD NOT", "RECOMMENDED", "NOT RECOMMENDED", "MAY", and "OPTIONAL" in this document are to be interpreted as described in RFC 2119 and RFC 8174.
+
+### Definitions
+
+- **Smart account** - An [ERC-4337](./eip-4337.md) compliant smart contract account that has a modular architecture.
+- **Module** - A smart contract with self-contained smart account functionality.
+  - Validator: A module used during the ERC-4337 validation flow to determine if a `UserOperation` is valid.
+  - Executor: A module that can execute transactions on behalf of the smart account via a callback.
+  - Fallback Handler: A module that can extend the fallback functionality of a smart account.
+- **Entrypoint** - A trusted singleton contract according to ERC-4337 specifications.
+
+### Account
+
+#### Validation
+
+This standard does not dictate how validator selection is implemented. However, should a smart account encode validator selection mechanisms in ERC-4337 `UserOperation` fields (i.e. `userOp.signature`), the smart account MUST sanitize the affected values before invoking the validator.
+
+The smart account's `validateUserOp` function SHOULD return the return value of the validator.
+
+#### Execution Behavior
+
+To comply with this standard, smart accounts MUST implement the two interfaces below. If an account implementation elects to not support any of the execution methods, it MUST implement the function, but revert instead of complying with the specifications below. This is in order to avoid unpredictable behavior with fallbacks.
+
+```solidity
+interface IExecution {
+    /**
+    * @dev packing batched transactions in a struct saves gas on both ERC-4337 and Executor Module flows
+    */
+    struct Execution {
+        address target;
+        uint256 value;
+        bytes callData;
+    }
+
+    /**
+    * MUST execute a `call` to the target with the provided data and value
+    * MUST allow ERC-4337 Entrypoint to be the sender and MAY allow `msg.sender == address(this)`
+    * MUST revert if the call was not successful
+    */
+    function execute(address target, uint256 value, bytes data) external returns (bytes memory result);
```

Comment:
change `s/data/callData`

Reply:
fixed in https://github.com/ethereum/ERCs/pull/163/commits/d0444089bc979c23960c4839a478894f08e579f1